From cb812119357d0a4df8b0ad8b40d62b788c77f2dd Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Thu, 25 Apr 2019 21:40:51 +0200
Subject: [PATCH 66/79] kernel/power: annotate {lock,unlock}_system_sleep

HANDLES:

kernel/power/main.c:28:1: warning: mutex 'system_transition_mutex' is still held at the end of function [-Wthread-safety-analysis]
}
^
kernel/power/main.c:27:2: note: mutex acquired here
        mutex_lock(&system_transition_mutex);
        ^
kernel/power/main.c:50:2: warning: releasing mutex 'system_transition_mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&system_transition_mutex);
        ^
---
 kernel/power/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/power/main.c b/kernel/power/main.c
index 98e76cad128b..51344a65ba1c 100644
--- a/kernel/power/main.c
+++ b/kernel/power/main.c
@@ -21,14 +21,14 @@
 
 #ifdef CONFIG_PM_SLEEP
 
-void lock_system_sleep(void)
+void lock_system_sleep(void) __attribute__((acquire_capability(system_transition_mutex)))
 {
 	current->flags |= PF_FREEZER_SKIP;
 	mutex_lock(&system_transition_mutex);
 }
 EXPORT_SYMBOL_GPL(lock_system_sleep);
 
-void unlock_system_sleep(void)
+void unlock_system_sleep(void) __attribute__((release_capability(system_transition_mutex)))
 {
 	/*
 	 * Don't use freezer_count() because we don't want the call to
-- 
2.17.1

